kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -296,36 +308,42 @@ def _init_and_run_proof(claim: KClaim) -> bool:
             smt_retry_limit=smt_retry_limit,
             trace_rewrites=trace_rewrites,
         ) as kcfg_explore:
-            proof_problem: APRProof
-
-            if save_directory is not None and not reinit and APRProof.proof_exists(claim.label, save_directory):
-                proof_problem = APRProof.read_proof(claim.label, save_directory)
-
+            proof_problem: Proof
+            if is_functional(claim):
+                proof_problem = EqualityProof.from_claim(claim, kevm.definition)
```

Comment:
We should still be checking if the proof state is already saved on disk here, rather than reloading it every time.

Basically teh same logic as below:

```suggestion
            proof_problem: Proof
            if is_functional(claim):
                if save_directory is not None and not reinit and EqualityProof.proof_exists(claim.label, save_directory):
                    proof_problem = EqualityProof.read_proof(claim.label, save_directory)
                else:
                    proof_problem = EqualityProof.from_claim(claim, kevm.definition)
```

Reply:
Since we are creating the proof with `EqualityProof.from_claim`, which always initializes `proof_dir` to `None`, they are never getting saved in the first place. I made a PR here:
https://github.com/runtimeverification/pyk/pull/475
which allows us to specify a save directory.

Reply:
This is fixed now and `EqualityProof`s are being written to files if requested.